biodatasets/mednli/mednli.py

```diff
@@ -0,0 +1,176 @@
+grounded in the medical history of patients. As the source of premise sentences, we used the
+MIMIC-III. More specifically, to minimize the risks to patient privacy, we worked with clinical
+notes corresponding to the deceased patients. The clinicians in our team suggested the Past Medical
+History to be the most informative section of a clinical note, from which useful inferences can be
+drawn about the patient.
+"""
+
+
+_HOMEPAGE = "https://physionet.org/content/mednli/1.0.0/"
+_LICENSE = "PhysioNet Credentialed Health Data License 1.5.0"
+
+_URLS = {}
+
+_SUPPORTED_TASKS = [Tasks.TEXTUAL_ENTAILMENT]
+
+_SOURCE_VERSION = "1.0.0"
+_BIGBIO_VERSION = "1.0.0"
+
+
+class MedNLIDataset(datasets.GeneratorBasedBuilder):
+    """MedNLI"""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    BUILDER_CONFIGS = [
+        BigBioConfig(
+            name="mednli_source",
+            version=SOURCE_VERSION,
+            description="MedNLI source schema",
+            schema="source",
+            subset_id="mednli",
+        ),
+        BigBioConfig(
+            name="mednli_bigbio_te",
+            version=BIGBIO_VERSION,
+            description="MedNLI BigBio schema",
+            schema="bigbio_te",
+            subset_id="mednli",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "mednli_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "pairID": datasets.Value("string"),
+                    "gold_label": datasets.Value("string"),
+                    "sentence1": datasets.Value("string"),
+                    "sentence2": datasets.Value("string"),
+                    "sentence1_parse": datasets.Value("string"),
+                    "sentence2_parse": datasets.Value("string"),
+                    "sentence1_binary_parse": datasets.Value("string"),
+                    "sentence2_binary_parse": datasets.Value("string"),
+                }
+            )
+
+        elif self.config.schema == "bigbio_te":
+            features = schemas.entailment_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
+        if self.config.data_dir is None:
+            raise ValueError("This is a local dataset. Please pass the data_dir kwarg to load_dataset.")
+        else:
+            data_dir = self.config.data_dir
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "mli_train_v1.jsonl"),
```

Comment:
in the other local dataset loaders we've tried to use the tar/zip/archive files directly. would it be possible for this script to work directly with the `mednli-a-natural-language-inference-dataset-for-the-clinical-domain-1.0.0.zip` file?